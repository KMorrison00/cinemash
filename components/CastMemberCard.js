import { View, Text, Image, StyleSheet } from "react-native"


export default function CastMemberCard({ castMember }) {
    // This code defines a functional component called CastMemberCard that takes a castMember object as a prop.
    const imageSrc = `https://image.tmdb.org/t/p/w500/${castMember.profile_path}`
    // It creates a URL for the cast member's profile picture using their profile_path property.
    return (
        <View style={styles.container}>
            <Image
                style={styles.profilePicture}
                source={{ uri: imageSrc }}
            />

            <Text style={styles.name}>
                {castMember.name}
            </Text>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        display: "flex",
        alignItems: "center",
        width: 100,
        height: 140,
    },
    name: {
        color: "#FFF",
        fontSize: 13,
    },
    profilePicture: {
        height: 100,
        width: 80,
        borderRadius: 5,
    }
})

