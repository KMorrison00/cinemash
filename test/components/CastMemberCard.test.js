import React from 'react';
import CastMemberCard from '../../components/CastMemberCard';
import renderer from 'react-test-renderer';

describe('CastMemberCard', () => {
    it('renders correctly', () => {
    const castMember = {
        profile_path: 'https://image.tmdb.org/t/p/w500/',
        name: 'John Doe',
    };

    const tree = renderer
        .create(<CastMemberCard castMember={castMember} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
    });
});
