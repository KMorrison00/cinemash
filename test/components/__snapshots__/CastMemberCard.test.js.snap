// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CastMemberCard renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "height": 140,
      "width": 100,
    }
  }
>
  <Image
    source={
      {
        "uri": "https://image.tmdb.org/t/p/w500/https://image.tmdb.org/t/p/w500/",
      }
    }
    style={
      {
        "borderRadius": 5,
        "height": 100,
        "width": 80,
      }
    }
  />
  <Text
    style={
      {
        "color": "#FFF",
        "fontSize": 13,
      }
    }
  >
    John Doe
  </Text>
</View>
`;
